packages/core/src/internal/newTask.js

```diff
@@ -39,15 +39,17 @@ export default function newTask(env, mainTask, parentContext, parentEffectId, me
       status = CANCELLED
       queue.cancelAll()
       // Ending with a TASK_CANCEL will propagate the Cancellation to all joiners
-      end(TASK_CANCEL)
+      end(TASK_CANCEL, false)
     }
   }
 
   function end(result, isErr) {
     if (!isErr) {
       // The status here may be RUNNING or CANCELLED
       // If the status is CANCELLED, then we do not need to change it here
-      if (status !== CANCELLED) {
+      if (result === TASK_CANCEL) {
+        status = CANCELLED
```

Comment:
Seems that this statement is unnecessary, see https://github.com/redux-saga/redux-saga/pull/1704#issuecomment-446208451

Reply:
Yeah, tests are passing because I havent add a new one (yet). It's true that `cancel` sets the status to `CANCELLED`, but `end` (as task.cont or mainTask.cont, cant remember right now) might be called from the proc too (without calling `task.cancel` here) when propagating the `TASK_CANCEL` through saga tree

Reply:
Yeah, I have thought there may be some similar cases that `task.end(TASK_CANCEL)` is called without calling `task.cancel()`,  but I failed to write a test case to prove the idea. It will be best you can add tests. 😄